### DISCUSSION TITLE
finding providers "conditionally" or by "priority"

### DISCUSSION THREAD

**Discussion by pseudobun** [wevm/mipd#13] [Ideas]
I think it would be nice to have the ability in [`findProvider`](https://github.com/wevm/mipd/blob/55bb0f66e2271bebe148379102ecba12b23e89e0/src/store.ts#L86)  method:

```js
findProvider({ rdns }) {
  return providerDetails.find(
    (providerDetail) => providerDetail.info.rdns === rdns,
  )
},
```

that would allow us to search for providers by "priority" or "conditionally".

I would suggest something like this:

```js
const provider = store.findProvider({
  rdns: 'io.metamask' || 'io.metamask.mmi' || 'io.metamask.flask' || 'com.enkrypt'
});
```

This way the dapps would be able to pick their "preferred" provider which in some cases would be great to have. E.g. we are developing a MetaMask Snap and therefore our preferred wallet is MetaMask since others don't even support Snaps.

Thanks for reply!

**Comment by jxom** [comment]
You can do this manually right now:

```ts
const providers = store.getProviders()
const provider = providers.find(provider => provider.info.rdns === 'io.metamask' || 'io.metamask.mmi' || 'io.metamask.flask' || 'com.enkrypt')
```

**Reply by pseudobun** [reply]
Completely overlooked this method. Thanks for such a quick reply!
